src/sentry/static/sentry/app/components/dataExport.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import {Client} from 'app/api';
+import Feature from 'app/components/acl/feature';
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import {Organization} from 'app/types';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+
+type DataExportPayload = {
+  queryType: number;
+  queryInfo: any; // TODO(ts): Formalize different possible payloads
+};
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  payload: DataExportPayload;
+};
+
+type State = {
+  inProgress: boolean;
+  dataExportId?: number;
+};
+
+const TooltipMessages = {
+  start: "We'll get all your data in one place and email you when it's ready",
+  progress: "We'll email you when it's ready",
+} as const;
+
+class DataExport extends React.Component<Props, State> {
+  state: State = {
+    inProgress: false,
+  };
+
+  async startDataExport() {
+    const {
+      api,
+      organization: {slug},
+      payload: {queryType, queryInfo},
+    } = this.props;
+    const {id: dataExportId} = await api.requestPromise(
```

Comment:
@leeandher Are you trying to set the `dataExportId` as the value of `id` through destructuring?

Reply:
Yeah sort of, the `id` is being destructured and then being renamed to `dataExportId` because its less generic, and matches state

Reply:
Cool